parsons/google/google_sheets.py

```diff
@@ -395,6 +399,148 @@ def format_cells(self, spreadsheet_id, range, cell_format, worksheet=0):
         ws.format(range, cell_format)
         logger.info("Formatted worksheet")
 
+    def attempt_gsheet_method(self, method, i=1, max=6, wait_time=15, **kwargs):
+        """
+        The Google Sheets API has notoriously strict rate limits (e.g. 60 calls per minute). This
+        function calls itself (i.e. is recursive) to help configure wait times and retry attempts
+        needed to wait out rate limit errors instead of letting them derail a script.
+        `Args:`
+            method: str
+                The name of the Parsons GoogleSheets method to be attempted
+            i: int
```

Comment:
Instead of exposing recursion control parameters to the public API, it is probably going to be cleaner to define an inner-function like this:

```python
def foo(a, b):
    def inner_foo(a, b, i):
         # <Do Recursive Stuff Here...>
    inner_foo(a, b, 0)
```